src/analyses/base.ml

```diff
@@ -2736,11 +2736,13 @@ struct
         let st = invalidate_ret_lv st in
         (* apply all registered abstract effects from other analysis on the base value domain *)
         LibraryFunctionEffects.effects_for f.vname args
-        |> List.map (fun sets ->
-            List.fold_left (fun acc (lv, x) ->
+        |> List.to_seq
+        |> Seq.map (fun sets ->
+          BatList.fold_left (fun acc (lv, x) ->
                 set ~ctx (Analyses.ask_of_ctx ctx) ctx.global acc (eval_lv (Analyses.ask_of_ctx ctx) ctx.global acc lv) (Cilfacade.typeOfLval lv) x
               ) st sets
           )
+        |> List.of_seq
         |> BatList.fold_left D.meet st
```

Comment:
Could one just use the `fold_left` on `Seq` here?

Reply:
Thanks for the tip, I tried it and it seems to work fine, so I will try to run Goblint now and see if we get any improvements.

Reply:
`Seq.map` and `Seq.fold_left` are fine, but one has to be careful using `Seq` functions, because many were just added in OCaml 4.14, while we also support 4.10: https://v2.ocaml.org/api/Seq.html.